packages/components/src/components/option/option.ts

```diff
@@ -0,0 +1,195 @@
+  }
+
+  private handleDefaultSlotChange() {
+    const textLabel = this.getTextLabel();
+
+    // Ignore the first time the label is set
+    if (typeof this.cachedTextLabel === 'undefined') {
+      this.cachedTextLabel = textLabel;
+      return;
+    }
+
+    // When the label changes, emit a slotchange event so parent controls see it
+    if (textLabel !== this.cachedTextLabel) {
+      this.cachedTextLabel = textLabel;
+      this.emit('slotchange', { bubbles: true, composed: false, cancelable: false });
+    }
+  }
+
+  private handleMouseEnter() {
+    this.hasHover = true;
+  }
+
+  private handleMouseLeave() {
+    this.hasHover = false;
+  }
+
+  @watch('disabled')
+  handleDisabledChange() {
+    this.setAttribute('aria-disabled', this.disabled ? 'true' : 'false');
+  }
+
+  @watch('selected')
+  handleSelectedChange() {
+    this.setAttribute('aria-selected', this.selected ? 'true' : 'false');
+  }
+
+  @watch('value')
+  handleValueChange() {
+    if (this.value.includes(' ')) {
+      console.error(`Option values cannot include a space. All spaces have been replaced with underscores.`, this);
+      this.value = this.value.replace(/ /g, '_');
+    }
+  }
+
+  /** Returns a plain text label based on the option's content. */
+  getTextLabel() {
+    return (this.textContent ?? '').trim();
+  }
+
+  render() {
+    // Slots
+    const slots = {
+      default: this.hasSlotController.test('default'),
+      left: this.hasSlotController.test('left'),
+      right: this.hasSlotController.test('right')
+    };
+
+    return html`
+      <div
+        part="base"
+        class=${cx(
+          'px-4 flex items-center w-full transition-all text-left text-base relative text-black',
+          { sm: 'py-1', md: 'py-2', lg: 'py-3' }[this.size],
+          { sm: 'text-sm', md: 'text-base', lg: 'text-base' }[this.size],
+          this.disabled ? 'text-neutral-500 cursor-not-allowed' : 'cursor-pointer',
+          this.hasHover && !this.disabled ? 'bg-neutral-200' : '',
+          this.current && 'bg-neutral-200'
+        )}
+        @mouseenter=${this.handleMouseEnter}
+        @mouseleave=${this.handleMouseLeave}
+      >
+        <span
+          class=${cx(
+            'absolute w-full left-0 top-0 pointer-events-none transition-all duration-150 border-l-4 h-[calc(100%-8px)]',
+            this.selected && !this.checkbox ? 'border-accent' : 'border-transparent',
+            this.disabled && 'border-neutral-500 top-1',
+            !this.disabled && this.hasHover ? 'h-full top-0' : 'top-1'
+          )}
+        ></span>
+        ${this.checkbox
+          ? html`<span
+              id="control"
+              part="control ${this.selected ? ' control--checked' : 'control--unchecked'}"
+              class=${cx(
+                'relative flex flex-initial items-center justify-center border rounded-sm h-4 w-4 mr-2',
+                this.disabled ? 'border-neutral-500' : this.selected ? 'bg-accent border-accent' : 'border-neutral-800'
```

Comment:
To improve readability, I would recommend the following:

```suggestion
                this.disabled 
                  ? 'border-neutral-500' 
                  : this.selected 
                    ? 'bg-accent border-accent' 
                    : 'border-neutral-800'
```